### ISSUE TITLE
This CPU lacks the 'UG' feature of EPT

### ISSUE THREAD

**Issue by suporterede1** [chyves/chyves#9]
I have computer  i5-4570 CPU @ 3.20GHz with VMware 6.0.
I installed a guest FreBSD (FreeBSD-10.3-STABLE-amd64-20160714-r302791-disc1) with bhyve but I have a error bellow.

`chyves upgrade
No pools configured.
This CPU lacks the 'UG' feature of EPT. This is required by bhyve to run to the full extent.
Due to this missing CPU feature: a guest is limited to one virtual CPU and starting UEFI guests is disabled.
See CPU feature flags below:
CPU: Intel(R) Core(TM) i5-4570 CPU @ 3.20GHz (3192.61-MHz K8-class CPU)
  Origin="GenuineIntel"  Id=0x306c3  Family=0x6  Model=0x3c  Stepping=3
  Features=0xfa3fbff<FPU,VME,DE,PSE,TSC,MSR,PAE,MCE,CX8,APIC,SEP,MTRR,PGE,MCA,CMOV,PAT,PSE36,DTS,MMX,FXSR,SSE,SSE2,SS>
  Features2=0xfffa3203<SSE3,PCLMULQDQ,SSSE3,FMA,CX16,PCID,SSE4.1,SSE4.2,x2APIC,MOVBE,POPCNT,TSCDLT,AESNI,XSAVE,OSXSAVE,AVX,F16C,RDRAND,HV>
  AMD Features=0x2c100800<SYSCALL,NX,Page1GB,RDTSCP,LM>
  AMD Features2=0x21<LAHF,ABM>
  Structured Extended Features=0x272a<TSCADJ,BMI1,AVX2,BMI2,ERMS,INVPCID,NFPUSG>
  XSAVE Features=0x1<XSAVEOPT>
  TSC: P-state invariant
Hypervisor: Origin = "VMwareVMware"
Run setup to install chyves dataset on a pool.
Critical error detected. Exiting for following reason:
No seriously, run 'chyves dataset <ZFS-pool-name> install`

**Comment by EpiJunkie** [comment]
Hello @suporterede1, thank you for raising an issue.

Could you format the command output with three grave accent (&#96;&#96;&#96;) symbols on the line before and after the command output? The important text in the &#60; &#62; are being interpreted as HTML tags.

As for the issue, my initial guess is that the VMware guest settings do not have the appropriate settings for nested hypervisors:

From the bhyve FAQ: https://wiki.freebsd.org/bhyve

> Q: Can I run multiple bhyve hosts under VMware nested VT-x EPT?
> 
> A: Yes, but [you must enable it](https://communities.vmware.com/docs/DOC-8970): look for the GUI or `vhv.allow` or `vhv.enable` options, depending on your version of VMware. VMware Fusion: Processors & Memory: Advanced options: Enable hypervisor applications in this virtual machine
> 
> Confirmed with VMware Workstation, VMware Player, VMware Fusion and ESXi. Note that you must stop all running VMs before you can enable EPT pass-through. 

Basically the CPU check in `chyves` looks for a line in the `dmesg` that looks like this (which contains the `UG` or unrestricted guest CPU feature):

```
  VT-x: PAT,HLT,MTF,PAUSE,EPT,UG,VPID
```

**Comment by suporterede1** [comment]
Hello

Now is ok. Dmesg show VTX.

`````` dmesg
Copyright (c) 1992-2016 The FreeBSD Project.
Copyright (c) 1979, 1980, 1983, 1986, 1988, 1989, 1991, 1992, 1993, 1994
        The Regents of the University of California. All rights reserved.
FreeBSD is a registered trademark of The FreeBSD Foundation.
FreeBSD 10.3-STABLE #0 r302791: Thu Jul 14 01:16:15 UTC 2016
    root@releng1.nyi.freebsd.org:/usr/obj/usr/src/sys/GENERIC amd64
FreeBSD clang version 3.4.1 (tags/RELEASE_34/dot1-final 208032) 20140512
CPU: Intel(R) Core(TM) i5-4570 CPU @ 3.20GHz (3192.61-MHz K8-class CPU)
  Origin="GenuineIntel"  Id=0x306c3  Family=0x6  Model=0x3c  Stepping=3
  Features=0xfa3fbff<FPU,VME,DE,PSE,TSC,MSR,PAE,MCE,CX8,APIC,SEP,MTRR,PGE,MCA,CMOV,PAT,PSE36,DTS,MMX,FXSR,SSE,SSE2,SS>
  Features2=0xfffa3223<SSE3,PCLMULQDQ,VMX,SSSE3,FMA,CX16,PCID,SSE4.1,SSE4.2,x2APIC,MOVBE,POPCNT,TSCDLT,AESNI,XSAVE,OSXSAVE,AVX                                                                                                               ,F16C,RDRAND,HV>
  AMD Features=0x2c100800<SYSCALL,NX,Page1GB,RDTSCP,LM>
  AMD Features2=0x21<LAHF,ABM>
  Structured Extended Features=0x272a<TSCADJ,BMI1,AVX2,BMI2,ERMS,INVPCID,NFPUSG>
  XSAVE Features=0x1<XSAVEOPT>
  VT-x: PAT,HLT,MTF,PAUSE,EPT,UG,VPID```


a) Power off guest.
Alter guest .vmx. Add line vhv.enable = "TRUE" and save file.

b) Alter option: Automatic to Use Intel VT-x/AMD-V for instruction set virtualization and Intel EPT/AMD RVI for MMU virtualization.
![guest-cpu mmu](https://cloud.githubusercontent.com/assets/22151195/18447869/d4a4b486-78fe-11e6-89e8-c5b8173d1128.jpg)

c) reboot guest.

``````

**Comment by EpiJunkie** [comment]
Great, thank you for the additional information.
